org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -964,6 +987,185 @@ public void recyclableThreads(ThreadReference thread) {
+            Value newValue = formatter.valueOf(value, type, options);
+            setValueFunc.apply(newValue);
+            return newValue;
+        }
+
+        private IValueFormatter getFormatterForModification(Type type, Map<String, Object> options) {
+            char signature0 = type.signature().charAt(0);
+
+            if (signature0 == LONG || signature0 == INT || signature0 == SHORT || signature0 == BYTE || signature0 == FLOAT
+                    || signature0 == DOUBLE || signature0 == BOOLEAN || signature0 == CHAR
+                    || type.signature().equals(STRING_SIGNATURE)) {
+                return this.variableFormatter.getValueFormatter(type, options);
+            }
+            throw new UnsupportedOperationException(String.format("Set value for type %s is not supported.", type.name()));
+        }
+
+        private Value setStaticFieldValue(Type declaringType, Field field, String name, String value, Map<String, Object> options)
+                throws ClassNotLoadedException, InvalidTypeException {
+            if (field.isFinal()) {
+                throw new UnsupportedOperationException(
+                        String.format("SetVariableRequest: Final field %s cannot be changed.", name));
+            }
+            if (!(declaringType instanceof ClassType)) {
+                throw new UnsupportedOperationException(
+                        String.format("SetVariableRequest: Field %s in interface cannot be changed.", name));
+            }
+            return setValueProxy(field.type(), value, newValue -> ((ClassType) declaringType).setValue(field, newValue), options);
+        }
+
+        private Value setFrameValue(StackFrame frame, LocalVariable localVariable, String value, Map<String, Object> options)
+                throws ClassNotLoadedException, InvalidTypeException {
+            return setValueProxy(localVariable.type(), value, newValue -> frame.setValue(localVariable, newValue), options);
+        }
+
+        private Value setObjectFieldValue(ObjectReference obj, Field field, String name, String value, Map<String, Object> options)
+                throws ClassNotLoadedException, InvalidTypeException {
+            if (field.isFinal()) {
+                throw new UnsupportedOperationException(
+                        String.format("SetVariableRequest: Final field %s cannot be changed.", name));
+            }
+            return setValueProxy(field.type(), value, newValue -> obj.setValue(field, newValue), options);
+        }
+
+        private Value setArrayValue(ArrayReference array, Type eleType, int index, String value, Map<String, Object> options)
+                throws ClassNotLoadedException, InvalidTypeException {
+            return setValueProxy(eleType, value, newValue -> array.setValue(index, newValue), options);
+        }
+
+        private Value setFieldValueWithConflict(ObjectReference obj, List<Field> fields, String name, String belongToClass,
```

Comment:
The name conflict can have the following scenarios:
1. the name conflict happens, but each name belongs to a different class name(after formatted).
2.  the name conflict happens, and unfortunately the  belonged to class name still conflicts, eg:
//Dup.java
package dup;
public class Dup {
    public  int x = 2;
}

//DupTest.java
package dup2;

class Dup extends dup.Dup {
    public  int x = 1;
}
public class DupTest extends Dup {
    public int x = 3;
}